lib/Auth/Source/OAuth2.php

```diff
@@ -213,6 +221,22 @@ function () use ($provider, $accessToken) {
         );
 
         $attributes = $resourceOwner->toArray();
+
+        foreach ($this->authenticatedApiRequests as $apiUrl) {
+            $apiAttributes = $this->retry(function () use ($provider, $accessToken, $apiUrl) {
+                /** @var Psr\Http\Message\RequestInterface $request */
+                $apiRequest = $provider->getAuthenticatedRequest(
+                    'GET',
+                    $apiUrl,
+                    $accessToken
+                );
+                return $provider->getParsedResponse($apiRequest);
+            },
+                $this->config->getInteger('retryOnError', 1)
+            );
+            $attributes = array_replace_recursive($attributes, $apiAttributes);
```

Comment:
The array structure of apiRequest may not be compatible with SSP attributes (e.g. it may have values that are maps). You need to call `convertResourceOwnerAttributes` to flatten the data structure.

Reply:
I think this is doen line 241.

Reply:
oh yeah, you're correct.